src/components/Global/ImageScaler.vue

```diff
@@ -0,0 +1,150 @@
+<script setup>
+import { ref, computed, onMounted, defineEmits, defineProps } from 'vue'
+import { useConfigurationStore } from '@/stores/configuration'
+import { fetchApiCall } from '@/utils/api'
+import RawScaledImage from './RawScaledImage.vue'
+
+// This component gets the raw image data from the server for an image
+// and then displays the image and the scaling controls under it
+
+const props = defineProps({
+  image: {
+    type: Object,
+    required: true
+  },
+  maxSize: {
+    type: Number,
+    default: 500
+  },
+  imageName: {
+    type: String,
+    required: true
+  },
+  compositeName: {
+    type: String,
+    required: true
+  }
+})
+
+const emit = defineEmits(['updateScaling'])
+const store = useConfigurationStore()
+const dataSessionsUrl = store.datalabApiBaseUrl
+const isLoading = ref(true)
+const errorReason = ref('')
+const rawData = ref({})
+const sliderRange = ref([0, 65535])
```

Comment:
Maybe we add a TODO to make this work with an arbitrary bit depth? The site "simple-stacked" images will be 32 bit.

Reply:
hooked in the bitdepth